src/MessagePack.UnityClient/Assets/Scripts/MessagePack/Resolvers/StaticCompositeResolver.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) All contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using MessagePack.Formatters;
+
+namespace MessagePack.Resolvers
+{
+    public class StaticCompositeResolver : IFormatterResolver
+    {
+        public static readonly StaticCompositeResolver Instance = new StaticCompositeResolver();
+
+        private bool freezed;
+        private IReadOnlyList<IMessagePackFormatter> formatters;
+        private IReadOnlyList<IFormatterResolver> resolvers;
+
+        private StaticCompositeResolver()
+        {
+            formatters = Array.Empty<IMessagePackFormatter>();
+            resolvers = Array.Empty<IFormatterResolver>();
+        }
+
+        public static void Register(params IMessagePackFormatter[] formatters)
```

Comment:
I think these `Register` methods merit an xml doc comment that explains that this *replaces* any previously registered formatters. Same for resolver overload.